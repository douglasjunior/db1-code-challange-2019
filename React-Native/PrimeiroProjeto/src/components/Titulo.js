import React from 'react';
import { Text } from 'react-native';

const Titulo = ({ textColor, children }) => {
    const style = {
        color: textColor,
        backgroundColor: 'yellow'
    };
    return (
        <Text style={style}>{children}</Text>
    );
};

export default Titulo;
