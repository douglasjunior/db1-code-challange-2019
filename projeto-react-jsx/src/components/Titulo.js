import React from 'react';

const Titulo = (props) => {
    const style = {
        color: props.textColor,
        backgroundColor: 'yellow'
    }

    return (
        <h1 style={style}>
          {props.children}
        </h1>
    );
};

export default Titulo;
