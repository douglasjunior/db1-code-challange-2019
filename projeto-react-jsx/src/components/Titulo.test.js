import React from 'react';
import testRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Titulo from './Titulo';

describe('Shallow tests', () => {
    test('Testa se o componente interno é um h1', () => {
        const component = shallow(<Titulo>Meu título</Titulo>);
        const element = component.getElement();
        expect(element.type).toBe('h1');
    });
    test('Testa a cor do texto', () => {
        const component = shallow(<Titulo textColor="green">Meu título</Titulo>);
        const styleColor = component.props().style.color;
        expect(styleColor).toBe('green');
    });
    test('Testa o texto de saída do componente', () => {
        const component = shallow(<Titulo textColor="green">Meu título</Titulo>);
        const text = component.text();
        expect(text).toBe('Meu título');
    });
});

describe('Snapshot tests', () => {
    test('Testa se o componente renderiza de acordo com a snapshot', () => {
        const tree = testRenderer
            .create(<Titulo>Meu título</Titulo>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
