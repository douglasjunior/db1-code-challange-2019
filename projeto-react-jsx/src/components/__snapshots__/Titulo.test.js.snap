// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Testa se o componente renderiza de acordo com a snapshot 1`] = `
<h1
  style={
    Object {
      "backgroundColor": "yellow",
      "color": undefined,
    }
  }
>
  Meu t√≠tulo
</h1>
`;
